### DISCUSSION TITLE
Incoming payment without an incomingAmount

### DISCUSSION THREAD

**Discussion by mkurapov** [interledger/rafiki#2885] [General]
### The current problems:
- the receiving ASE does not know the maximum amount the wallet address will receive. The ASE is unable to respond to a particular amount of an incoming payment
- payments do not get completed unless they expire or manually completed. Having to complete an incoming payment manually adds another request for the OP client (and in which scenarios does it actually need to be completed?), and setting a short expiry is gambling: either the payment expires before a funds can be transferred or the receiver needs to wait longer than they have to finally get their funds, even though the full amount of the outgoing payment was sent.

### Proposal

The receiver during quote creation can be an incoming payment **or a wallet address.**

If a wallet address is provided, then under the hood, we have the ILP rate probe send unfulfillable packets to a wallet address over SPSP, by which we are able to get the receive amount.

After the rate probe, Rafiki under the hood creates an incoming payment on the wallet address with an exact incomingAmount, which is equal to the minimumDelivery amount of the ILP quote.

This also saves the ASE a createReceiver call when doing P2P payments.
They just need to: createQuote and createOutgoingPayment , and once the outgoing payment is completed, the incoming payment will automatically be as well, since it knows the exact amount it needs to receive

At first we can add this functionality to Rafiki only for the Admin API, and if we want to, we can also add it to Open Payments as well

_TL;DR during quoting, we send unfulfillable packet to the SPSP details for the wallet address to get the exchange rate first, then Rafiki creates the incoming payment on the receiver with the quoted amount._
